docs/content/en/docs/pipeline-stages/lifecycle-hooks.md

```diff
@@ -18,6 +18,10 @@ Host hooks are executed on the runner and can be defined for the following phase
 
 ### `before-build` and `after-build`
 
+Build hooks are executed if the build phase runs. If the resulting artifact
+is cached, the hooks are not executed. To test the build hooks, run Skaffold
+with `--cache-artifacts=false` option.
```

Comment:
This text implies that build hooks are run for the _phase_, rather than for each artifact.

```suggestion
Build hooks are executed before and after each artifact is built. 
If an artifact is not built, such as happens when the image was found in the Skaffold image cache, then the build hooks will not be executed.
To force the build hooks, run Skaffold with `--cache-artifacts=false` option.
```